taxonomy-editor-frontend/src/pages/project/search/MultipleSelectFilter.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  FormControl,
+  OutlinedInput,
+  Checkbox,
+  Select,
+  MenuItem,
+  InputLabel,
+  ListItemText,
+} from "@mui/material";
+import { ChangeEvent, Dispatch, SetStateAction } from "react";
+
+type MultipleSelectFilterType = {
+  label: string;
+  filterValue: string[];
+  listOfChoices: string[];
+  mapCodeToValue: (code: string) => string;
+  mapValueToCode: (value: string) => string;
+  setQ: Dispatch<SetStateAction<string>>;
+  keySearchTerm: string;
+  setCurrentPage: Dispatch<SetStateAction<number>>;
+};
+
+export const MultipleSelectFilter = ({
+  label,
+  filterValue,
+  listOfChoices,
+  mapCodeToValue = () => "",
+  mapValueToCode = () => "",
+  setQ,
+  keySearchTerm,
+  setCurrentPage,
+}: MultipleSelectFilterType) => {
+  const handleChange = (
+    event: ChangeEvent<HTMLInputElement>,
+    languageCodeItem: string
+  ) => {
+    setCurrentPage(1);
+    if (!filterValue.includes(languageCodeItem)) {
+      setQ((prevQ) => prevQ + ` ${keySearchTerm}:${languageCodeItem}`);
+    } else {
+      setQ((prevQ) =>
+        prevQ.replace(`${keySearchTerm}:${languageCodeItem}`, "")
+      );
+    }
+    event.target.closest("Menu")?.dispatchEvent(new Event("close"));
+  };
+  return (
+    <FormControl sx={{ m: 1 }}>
+      <InputLabel id="multiple-select-label">{label}</InputLabel>
+      <Select
+        id="languages-filter"
+        sx={{ width: "170px" }}
+        multiple
+        value={filterValue}
+        input={<OutlinedInput label="Languages" />}
+        renderValue={(selected) =>
+          selected
+            .map((langCode) => mapCodeToValue(langCode))
+            .filter(Boolean) //to ignore "xx" language
+            .join(", ")
+        }
+      >
+        {listOfChoices.sort().map((languageNameItem) => {
```

Comment:
```suggestion
        {listOfChoices.map((languageNameItem) => {
```
I think that the list should be sorted in the parent component if necessary